framework/elements/BaseElement.js

```diff
@@ -0,0 +1,107 @@
+import { Timeouts } from '../../constants/timeouts'
+
+class BaseElement {
+  constructor(name, selector) {
+    this.name = name
+    this.selector = selector
+  }
+
+  async getElement() {
+    return await $(this.selector)
+  }
+
+  async waitForExist(timeout, reverse = false, interval) {
+    try {
+      await this.getElement().waitForExist({ timeout, reverse, interval: interval ?? Timeouts.WAIT_FOR_INTERVAL })
+    } catch (error) {
+      console.error('Error waiting for element to exist:', error)
+      throw error
+    }
+  }
+
+  async waitForDisplayed(timeout, reverse = false, interval) {
+    try {
+      await this.getElement().waitForDisplayed({ timeout, reverse, interval: interval ?? Timeouts.WAIT_FOR_INTERVAL })
+    } catch (error) {
+      console.error('Error waiting for element to be displayed:', error)
+      throw error
+    }
+  }
+
+  async waitForClickable(timeout, reverse = false, interval) {
+    try {
+      await this.getElement().waitForClickable({ timeout, reverse, interval: interval ?? Timeouts.WAIT_FOR_INTERVAL })
+    } catch (error) {
+      console.error('Error waiting for element to be clickable:', error)
+      throw error
+    }
+  }
+
+  async waitForEnabled(timeout, reverse = false, interval) {
+    try {
+      await this.getElement().waitForEnabled({ timeout, reverse, interval: interval ?? Timeouts.WAIT_FOR_INTERVAL })
+    } catch (error) {
+      console.error('Error waiting for element to be enabled:', error)
+      throw error
+    }
+  }
+
+  async isExisting(timeout) {
+    return await this.getElement().isExisting({ timeout })
+  }
+
+  async isDisplayed(timeout) {
```

Comment:
I meant something like that:
![image](https://github.com/teiah/vention-internship/assets/98515302/388b7bb7-0947-45c6-ae70-0bcd2485c0d6)